include/ofxPropsMetadata.h

```diff
@@ -0,0 +1,224 @@
+// Copyright OpenFX and contributors to the OpenFX project.
+// SPDX-License-Identifier: BSD-3-Clause
+// NOTE: This file is auto-generated by gen-props.py. DO NOT EDIT.
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include "ofxImageEffect.h"
+#include "ofxGPURender.h"
+#include "ofxColour.h"
+#include "ofxDrawSuite.h"
+#include "ofxParametricParam.h"
+#include "ofxKeySyms.h"
+#include "ofxOld.h"
+
+namespace OpenFX {
+enum class PropType {
+   Int,
+   Double,
+   Enum,
+   Bool,
+   String,
+   Bytes,
+   Pointer
+};
+
+enum class Writable {
+   Host,
+   Plugin,
+   All
+};
+
+struct PropsMetadata {
+  std::string name;
+  std::vector<PropType> types;
+  int dimension;
+  Writable writable;
+  bool host_optional;
+  std::vector<const char *> values; // for enums
+};
+
+const std::vector<struct PropsMetadata> props_metadata {
```

Comment:
Since this it auto generated, you could consider `std::array<struct PropsMetadata, N=FeedByScrip>`.

If this header is intended to be use more C++, you should consider having this `std::array` as constexpr.

maybe:

```
#if defined(__cplusplus)
#define CONST_SPECIFIER constexpr
#else
#define CONST_SPECIFIER const
#endif
```

Reply:
Thanks @abizeauMaxon -- totally agreed. As per discussion at the TSC today, I've moved these into `Support/include` and they are 100% C++. I used fixed-size `std::array`s where possible, as well as some other updates and cleanups. More feedback will be helpful; let me know if you try it out.